### PR TITLE
[BE] Make `torch.cuda.has_magma` a build time check

### PR DESCRIPTION
**Pull request by malfet**
Perhaps originally one needed to query about GPU capability, but right now it's a simple check for a build time flag: https://github.com/pytorch/pytorch/blob/52f0457d7dc8e1ee6e25b6c97d11d3070d11263b/aten/src/ATen/cuda/detail/CUDAHooks.cpp#L165-L171

Alternative, to avoid `at::hasMAGMA()` call  one can implement it as follows:
```cpp
  const auto use_magma = caffe2::GetBuildOptions().at("USE_MAGMA");
  return PyBool_FromLong(use_magma == "1");
```

Make this check very similar to `_has_mkldnn`
https://github.com/pytorch/pytorch/blob/0978482afa69118da1a986a4edec3acea01d2c6d/torch/csrc/Module.cpp#L1793-L1794

Test plan:
 Run `lldb -- python3 -c "import torch;print(torch.cuda.has_magma)"` and make sure it returns True and that `cuInit` is not called

### CODE REVIEW

**Review by seemethere — APPROVED** [review]

**Comment by malfet** [comment]
@pytorch merge

File: torch/csrc/cuda/Module.cpp [hunk 1]
```diff
@@ -873,6 +873,12 @@ PyObject* THCPModule_cudaGetSyncDebugMode(PyObject* self, PyObject* noargs) {
   END_HANDLE_TH_ERRORS
 }
 
+static PyObject* THCPModule_hasMagma(PyObject* self, PyObject* noargs) {
+  HANDLE_TH_ERRORS
+  return at::hasMAGMA() ? Py_True : Py_False;
```
Comment (albanD) [inline torch/csrc/cuda/Module.cpp:878]: This is missing refcount bumps!
Py_RETURN_TRUE and cie needs to be used

File: torch/cuda/__init__.py [hunk 2]
```diff
@@ -111,8 +111,8 @@ def _maybe_exchange_device(device: int) -> int:
 
 
 has_half: bool = True
-# Global variables dynamically populated by native code
-has_magma: bool = False
+has_magma: bool = getattr(torch._C, "_cuda_hasMagma", lambda: False)()
```
Comment (albanD) [inline torch/cuda/__init__.py:114]: This is meh
You can put this function in the general binding (instead of the cuda-only one) if you always need it.

**Review by albanD — CHANGES_REQUESTED** [review]

File: torch/cuda/__init__.py (hunk 2 above)
Comment (malfet) [reply torch/cuda/__init__.py:114]: I was about to object, that Module.cpp should export no CUDA specific symbols, but as it exports `_has_cuda`, then I don't see why not do the same for magma

File: torch/csrc/cuda/Module.cpp (hunk 1 above)
Comment (malfet) [reply torch/csrc/cuda/Module.cpp]: I've moved the code to `Module.cpp`, so it still returns `Py_True` but it should not matter, should it?

**Review by albanD — APPROVED** [review]
Sounds good!

**Pull request closed without merging**